lib/rules/async-server-action.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @fileoverview Require functions with the `use server` directive to be async
+ * @author Jorge Zreik
+ */
+
+'use strict';
+
+const docsUrl = require('../util/docsUrl');
+const report = require('../util/report');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+const messages = {
+  asyncServerAction: 'Your server action should be async',
+};
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Require functions with the `use server` directive to be async',
+      category: 'Possible Errors',
+      recommended: false,
+      url: docsUrl('async-server-action'),
+    },
+
+    messages,
+
+    fixable: 'code',
+
+    schema: [],
+  },
+
+  create(context) {
+    let currentFunction = null;
+
+    return {
+      ':function'(node) {
+        currentFunction = node;
+      },
+      ':function[async=false]>BlockStatement>:first-child[expression.value="use server"]'() {
+        report(context, messages.asyncServerAction, 'asyncServerAction', {
+          node: currentFunction,
```

Comment:
```suggestion
      ':function[async=false]>BlockStatement>:first-child[expression.value="use server"]'(node) {
        report(context, messages.asyncServerAction, 'asyncServerAction', {
          node: node.parentNode.parentNode,
```
or something like this? then you wouldn't need the `:function` listener at all.

Reply:
Yeah that works for sure. I felt the listener was clearer but just chaining up parents is definitely simpler. Changed.